timesketch/lib/sigma.py

```diff
@@ -0,0 +1,193 @@
+
+    if not config_file:
+        raise ValueError(
+            'SIGMA_CONFIG not found in config file')
+
+    if not os.path.isfile(config_file):
+        raise ValueError(
+            'Unable to open file: [{0:s}], it does not exist.'.format(
+                config_file))
+
+    if not os.access(config_file, os.R_OK):
+        raise ValueError(
+            'Unable to open file: [{0:s}], cannot open it for '
+            'read, please check permissions.'.format(config_file))
+
+    with open(config_file, 'r') as config_file:
+        sigma_config_file = config_file.read()
+
+    sigma_config = sigma_configuration.SigmaConfiguration(sigma_config_file)
+
+    return sigma_config
+
+def get_sigma_rules_path():
+    """Get Sigma rules paths.
+
+        Returns:
+            A list of strings to the Sigma rules
+    """
+    rules_path = current_app.config.get('SIGMA_RULES_FOLDERS', [])
+
+    if not rules_path:
+        raise ValueError(
+            'SIGMA_RULES_FOLDERS not found in config file')
+
+    for folder in rules_path:
+        if not os.path.isdir(folder):
+            raise ValueError(
+                'Unable to open dir: [{0:s}], it does not exist.'.format(
+                    folder))
+
+        if not os.access(folder, os.R_OK):
+            raise ValueError(
+                'Unable to open dir: [{0:s}], cannot open it for '
+                'read, please check permissions.'.format(folder))
+
+    return rules_path
+
+
+def get_sigma_rules(rule_folder):
+    """Returns the Sigma rules for a folder including subfolders.
+
+        Args:
+            rule_folder: folder to be checked for rules
+        Returns:
+            A array of Sigma rules as JSON
+    """
+    return_array = []
+
+    for dirpath, dirnames, files in os.walk(rule_folder):
+        if 'deprecated' in [x.lower() for x in dirnames]:
+            dirnames.remove('deprecated')
+
+        for rule_filename in files:
+            if rule_filename.lower().endswith('.yml'):
+                # if a sub dir is found, do not try to parse it.
+                if os.path.isdir(os.path.join(dirpath, rule_filename)):
+                    continue
+
+                rule_file_path = os.path.join(dirpath, rule_filename)
+                parsed_rule = get_sigma_rule(rule_file_path)
+                if parsed_rule:
+                    return_array.append(parsed_rule)
+
+    return return_array
+
+def get_all_sigma_rules():
+    """Returns all Sigma rules
+
+    Returns:
+        A array of Sigma rules
+    """
+
+    sigma_rules = []
+
+    rules_paths = get_sigma_rules_path()
+
+    for folder in rules_paths:
+        sigma_rules.extend(get_sigma_rules(folder))
+
+    return sigma_rules
+
+
+def get_sigma_rule(filepath):
+    """ Returns a JSON represenation for a rule
+
+        Args:
+            filepath: path to the sigma rule to be parsed
+
+        Returns:
+            Json representation of the parsed rule
+        """
```

Comment:
doc string indentation...

Reply:
was actually off in the whole file, fixed